lib/config_files/grub2/default.rb

```diff
@@ -0,0 +1,176 @@
+      end
+
+      def os_prober
+        @os_prober ||= BooleanValue.new(
+          "GRUB_DISABLE_OS_PROBER",
+          self,
+          default:     true,
+          # grub key is disable, so use reverse logic
+          true_value:  "false",
+          false_value: "true"
+        )
+      end
+
+      def disable_recovery_entry
+        generic_set("GRUB_DISABLE_RECOVERY", "true")
+      end
+
+      def enable_recovery_entry(kernel_params)
+        generic_set("GRUB_DISABLE_RECOVERY", "false")
+        generic_set("GRUB_CMDLINE_LINUX_RECOVERY", kernel_params)
+      end
+
+      def timeout
+        data["GRUB_TIMEOUT"] || 10
+      end
+
+      def timeout=(value)
+        generic_set("GRUB_TIMEOUT", value)
+      end
+
+      def cryptodisk
+        @cryptodisk ||= BooleanValue.new(
+          "GRUB_ENABLE_CRYPTODISK",
+          self
+        )
+      end
+
+      def terminal
+        case data["GRUB_TERMINAL"]
+        when "console", "", nil
+          :console
+        when "serial"
+          :serial
+        when "gfxterm"
+          :gfxterm
+        else
+          raise "unknown GRUB_TERMINAL option #{data["GRUB_TERMINAL"].inspect}"
+        end
+      end
+
+      VALID_TERMINAL_OPTIONS = [:serial, :console, :gfxterm]
+      def terminal=(value)
+        if !VALID_TERMINAL_OPTIONS.include?(value)
+          raise ArgumentError, "invalid value #{value.inspect}"
+        end
+
+        generic_set("GRUB_TERMINAL", value.to_s)
+      end
+
+      def serial_console=(value)
+        self.terminal = :serial
+        generic_set("GRUB_SERIAL_COMMAND", value)
+      end
+
+      def serial_console
+        data["GRUB_SERIAL_COMMAND"]
+      end
+
+      # powerfull low level method that sets any value in grub config.
+      # @note prefer to use specialized methods
+      def generic_set(key, value)
+        modify(key, value) || uncomment(key, value) || add_new(key, value)
+      end
+
+      # powerfull method that gets unformatted any value in grub config.
+      # @note prefer to use specialized methods
+      def generic_get(key)
+        data[key]
+      end
+
+    private
+
+      def modify(key, value)
+        # if already set, just change value
+        return false unless data[key]
+
+        data[key] = value
+        true
+      end
+
+      def uncomment(key, value)
+        # Try to find if it is commented out, so we can replace line
+        matcher = AugeasMatcher.new(
+          collection:    "#comment",
+          value_matcher: /#{key}\s*=/
+        )
+        return false unless  data.data.any?(&matcher)
+
+        data.add(key, value, AugeasReplacePlacer.new(matcher))
+        true
+      end
+
+      def add_new(key, value)
+        data.add(key, value)
+      end
+
+      # Representing boolean value switcher in default grub configuration file.
+      # Allows easy switching and questioning for boolean value, even if
+      # represented by text in config file
+      class BooleanValue
```

Comment:
Same than `generic_set`. I guess this is here just temporarily, because it really looks like something generic library-wide.